cmd/flightctl-agent/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/flightctl/flightctl/internal/agent"
+	"github.com/flightctl/flightctl/internal/agent/controller"
+	"k8s.io/klog/v2"
+)
+
+func main() {
+	machineInfo, err := controller.GetSystemInfo()
+	if err != nil {
+		log.Fatalf("getting system info: %v", err)
+	}
+	serverUrl := flag.String("server", "", "device server URL")
+	dataDir := flag.String("data-dir", "/var/lib/flightctl", "device agent data directory")
+
+	// TODO: switch to a better default, i.e. IDevID PK hash from TPM
+	// https://trustedcomputinggroup.org/resource/tpm-2-0-keys-for-device-identity-and-attestation/
+	agentName := flag.String("agent-name", "agent-"+machineInfo.MachineID, "device agent name")
+
+	fetchSpecInterval := flag.Duration("fetch-spec-interval", agent.DefaultFetchSpecInterval, "Duration between two reads of the remote device spec")
+	statusUpdateInterval := flag.Duration("status-update-interval", agent.DefaultStatusUpdateInterval, "Duration between two status updates")
+	flag.Parse()
+
+	klog.Infoln("starting flightctl device agent")
+	defer klog.Infoln("flightctl device agent stopped")
+
+	klog.Infoln("command line flags:")
+	flag.CommandLine.VisitAll(func(flg *flag.Flag) {
+		klog.Infof("  %s=%s", flg.Name, flg.Value)
+	})
+
+	if *serverUrl == "" {
+		klog.Fatalf("flightctl server URL is required")
+	}
+
+	agentInstance := agent.NewDeviceAgent(*serverUrl, *serverUrl, *dataDir).
+		SetName(*agentName).
+		AddController(controller.NewSystemInfoController()).
+		SetFetchSpecInterval(*fetchSpecInterval, 0).
+		SetStatusUpdateInterval(*statusUpdateInterval, 0)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	sigShutdown := make(chan os.Signal, 1)
+	signal.Notify(sigShutdown, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		sig := <-sigShutdown
+		signal.Stop(sigShutdown)
+		log.Printf("Shutdown signal received (%v).", sig)
+		cancel()
+	}()
+
+	agentInstance.Run(ctx)
```

Comment:
Add error reporting here. In fact I remember doing this, so I probably haven't sent last changes.